lib/kubeclient/watch_stream.rb

```diff
@@ -4,10 +4,11 @@ module Kubeclient
   module Common
     # HTTP Stream used to watch changes on entities
     class WatchStream
-      def initialize(uri, options)
+      def initialize(uri, http_options, format: :json)
         @uri = uri
         @http = nil
-        @options = options.merge(read_timeout: nil)
+        @options = http_options.merge(read_timeout: nil)
```

Comment:
@jonmoter at this point it's probably worth renaming also `@options` to `@http_options`